File: src/main/java/CaesarCipher/Caesarcipher.java
package CaesarCipher;


public class Caesarcipher {
    private final String ALPHABET=" ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    private  String EncryptMessages(String message,int key)
    {
        String CipherText="";
        message=message.toUpperCase();
        for (int i = 0; i < message.length(); i++) {
            char character=message.charAt(i);
            int charIndex=ALPHABET.indexOf(character);
            int encryptedIndex=(charIndex+key)%ALPHABET.length();
            CipherText+=ALPHABET.charAt(encryptedIndex);
        }
        return CipherText;
    }
    private  String DecryptMessages(String message,int key)
    {
        String CipherText="";
        message=message.toUpperCase();
        for (int i = 0; i < message.length(); i++) {
            char character=message.charAt(i);
            int charIndex=ALPHABET.indexOf(character);
            int encryptedIndex=(charIndex-key)%ALPHABET.length();
            CipherText+=ALPHABET.charAt(encryptedIndex);
        }
        return CipherText;
    }
    public static void main(String[] args) {
        System.out.println(new Caesarcipher().EncryptMessages("ABC",5));
        System.out.println(new Caesarcipher().DecryptMessages("FGH",5));
    }
}

File: src/main/python/CaesarCipher/CaesarCipher.py
ALPHABET=' ABCDEFGHIJKLMNOPQRSTUVWXYZ' #27
class CaesarCipher:
    def __init__(self) -> None:
        pass
    def caesar_encryption(message,key):
        cipher_text=''
        message=message.upper()
        for i in message:
            index=ALPHABET.find(i)
            index=(index+key)%len(ALPHABET)
            cipher_text+=ALPHABET[index];
        return cipher_text
    def caesar_decryption(message,key):
        cipher_text=''
        for i in message:
            index=ALPHABET.find(i)
            index=(index-key)%len(ALPHABET)
            cipher_text+=ALPHABET[index]
        return cipher_text
if __name__=='__main__':
    encrypt=CaesarCipher.caesar_encryption("I LOVE YOU",5)
    print(encrypt)
    decrypt=CaesarCipher.caesar_decryption(encrypt,5)
    print(decrypt)
